haystack/components/preprocessors/markdown_header_splitter.py

```diff
@@ -0,0 +1,405 @@
+import logging
+import re
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
```

Comment:
Let's make sure to use the logging from Haystack
```suggestion
import re
from typing import Any, Dict, List, Literal, Optional

from haystack import Document, component, default_from_dict, default_to_dict, logging
```

Reply:
You'll also need to update how the logger statements are called. For example, this
```python
logger.warning(
    f"Agent reached maximum agent steps of {max_agent_steps}, stopping."
)
```
would be come this
```python
logger.warning(
    "Agent reached maximum agent steps of {max_agent_steps}, stopping.",
    max_agent_steps=self.max_agent_steps,
)
```